### PR TITLE
Drop `attrs` dependency and support for Python 3.6; refactor to use stdlib `dataclasses`

### PR DESCRIPTION
**Pull request by AlexWaygood**
Refs #63

### CODE REVIEW

File: pyi.py [hunk 1]
```diff
@@ -438,14 +437,14 @@ def run(self, tree: ast.AST) -> Iterable[Error]:
...
-    version = __version__
+    name: ClassVar[str] = "flake8-pyi"
+    version: ClassVar[str] = __version__
 
-    tree = attr.ib(default=None)
-    filename = attr.ib(default="(none)")
-    options = attr.ib(default=None)
+    tree: Optional[ast.Module] = None
+    filename: str = "(none)"
+    options: Any = None
```
Comment (AlexWaygood) [inline pyi.py:447]: I... wasn't actually sure what the proper type annotation for this was

Comment (Akuli) [reply pyi.py:447]: I think `Any` is fine for now, but if you want, you can add `print(self.options)` somewhere and run the tests.

Comment (AlexWaygood) [reply pyi.py]: Seems like it's an `argparse.Namespace`.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -9,6 +9,7 @@ unreleased
 * extend Y001 to cover ParamSpec and TypeVarTuple in addition to TypeVar
 * introduce Y016 (duplicate union member)
 * support Python 3.10
+* discontinue support for 3.6
```
Comment (Akuli) [inline CHANGELOG.rst:12]:
```suggestion
* discontinue support for Python 3.6
```

File: pyi.py [hunk 3]
```diff
@@ -292,7 +292,6 @@ def _check_subscript_version_check(self, node: ast.Compare) -> None:
                 # Python 3.9 flattens the AST and removes Index, so simulate that here
                 slice_num = slc if isinstance(slc, ast.Num) else slc.value
                 # anything other than the integer 0 doesn't make much sense
-                # (things that are in 2.7 and 3.7 but not 3.6?)
                 if isinstance(slice_num, ast.Num) and slice_num.n == 0:
```
Comment (Akuli) [inline pyi.py:295]: This code has a potential bug, similar to #60, because `0.0 == 0`. We should check if that can be a problem.

File: setup.py [hunk 4]
```diff
@@ -46,7 +46,6 @@
         "License :: OSI Approved :: MIT License",
         "Operating System :: OS Independent",
         "Programming Language :: Python",
-        "Programming Language :: Python :: 3.6",
```
Comment (Akuli) [inline setup.py:49]: `setup.cfg` still says py36. I'm not sure if we even need that file. (I ran `git grep 6` locally)

Comment (AlexWaygood) [reply setup.py:49]: Nor do I, but I've fixed it anyhow :)

**Review by Akuli — APPROVED** [review]

**Pull request merged**